test/forge/RevokeTest.sol

```diff
@@ -0,0 +1,141 @@
+        super.setUp();
+
+        vm.prank(OWNER);
+        vault.setFeeRecipient(FEE_RECIPIENT);
+
+        _setFee(FEE);
+        _setTimelock(TIMELOCK);
+        _setGuardian(GUARDIAN);
+    }
+
+    function testOwnerRevokeTimelockDecreased(uint256 timelock, uint256 elapsed) public {
+        timelock = bound(timelock, ConstantsLib.MIN_TIMELOCK, TIMELOCK - 1);
+        elapsed = bound(elapsed, 0, TIMELOCK - 1);
+
+        vm.prank(OWNER);
+        vault.submitTimelock(timelock);
+
+        vm.warp(block.timestamp + elapsed);
+
+        vm.expectEmit();
+        emit EventsLib.RevokeTimelock(OWNER, IPending(address(vault)).pendingTimelock());
+        vm.prank(OWNER);
+        vault.revokeTimelock();
+
+        uint256 newTimelock = vault.timelock();
+        (uint256 pendingTimelock, uint64 submittedAt) = vault.pendingTimelock();
+
+        assertEq(newTimelock, TIMELOCK, "newTimelock");
+        assertEq(pendingTimelock, 0, "pendingTimelock");
+        assertEq(submittedAt, 0, "submittedAt");
+    }
+
+    function testCuratorRevokeCapIncreased(uint256 seed, uint256 cap, uint256 elapsed) public {
+        MarketParams memory marketParams = _randomMarketParams(seed);
+        elapsed = bound(elapsed, 0, TIMELOCK - 1);
+        cap = bound(cap, 1, type(uint192).max);
+
+        vm.prank(OWNER);
+        vault.submitCap(marketParams, cap);
+
+        vm.warp(block.timestamp + elapsed);
+
+        Id id = marketParams.id();
+
+        vm.expectEmit();
+        emit EventsLib.RevokeCap(CURATOR, id, IPending(address(vault)).pendingCap(id));
+        vm.prank(CURATOR);
+        vault.revokeCap(id);
+
+        (uint192 newCap, uint64 withdrawRank) = vault.config(id);
+        (uint256 pendingCap, uint64 submittedAt) = vault.pendingCap(id);
+
+        assertEq(newCap, 0, "newCap");
+        assertEq(withdrawRank, 0, "withdrawRank");
+        assertEq(pendingCap, 0, "pendingCap");
+        assertEq(submittedAt, 0, "submittedAt");
+    }
+
+    function testOwnerRevokeCapIncreased(uint256 seed, uint256 cap, uint256 elapsed) public {
+        MarketParams memory marketParams = _randomMarketParams(seed);
+        elapsed = bound(elapsed, 0, TIMELOCK - 1);
+        cap = bound(cap, 1, type(uint192).max);
+
+        vm.prank(OWNER);
+        vault.submitCap(marketParams, cap);
+
+        vm.warp(block.timestamp + elapsed);
+
+        Id id = marketParams.id();
+
+        vm.expectEmit();
+        emit EventsLib.RevokeCap(OWNER, id, IPending(address(vault)).pendingCap(id));
+        vm.prank(OWNER);
+        vault.revokeCap(id);
+
+        (uint192 newCap, uint64 withdrawRank) = vault.config(id);
+        (uint256 pendingCap, uint64 submittedAt) = vault.pendingCap(id);
+
+        assertEq(newCap, 0, "newCap");
+        assertEq(withdrawRank, 0, "withdrawRank");
+        assertEq(pendingCap, 0, "pendingCap");
+        assertEq(submittedAt, 0, "submittedAt");
+    }
+
+    function testOwnerRevokeGuardian(uint256 elapsed) public {
```

Comment:
The owner MUST NOT be able to do this

Reply:
This is incorrect

Reply:
No in fact it's fine

[This is cantina's suggestion](https://github.com/cantinasec/review-morpho-blue-1/issues/92)